devicetree error: failed to handle raspberry Pico connector

### Describe the bug

I'm working on a project with Raspberry Pico and WiFi is added by introducing Nordic nRF7002. While Arduino  connector should be handed first. FlexyPin Adapter is what I use to do it.

It looks like rpi_pico_uno_flexypin.overlay isn't compatible with rpi_pico-common.dtsi. I have reviewed these two files and got nowhere.

### Regression

- [ ] This is a regression.

### Steps to reproduce

west build -b rpi_pico --shield rpi_pico_uno_flexypin --shield nrf7002ek

In my case, blinky is used to reproduce.

### Relevant log output

```shell
west build -b rpi_pico --shield rpi_pico_uno_flexypin --shield nrf7002ek
-- west build: generating a build system
Loading Zephyr default modules (Zephyr base).
-- Application: /home/freddie/Workspaces/zephyrproject/examples/blinky
-- CMake version: 3.28.3
-- Found Python3: /home/freddie/Workspaces/zephyrproject/.venv/bin/python3 (found suitable version "3.12.3", minimum required is "3.10") found components: Interpreter 
-- Cache files will be written to: /home/freddie/.cache/zephyr
-- Zephyr version: 4.2.99 (/home/freddie/Workspaces/zephyrproject/zephyr)
-- Found west (found suitable version "1.5.0", minimum required is "0.14.0")
-- Board: rpi_pico, qualifiers: rp2040
-- Shield(s): rpi_pico_uno_flexypin;nrf7002ek
-- ZEPHYR_TOOLCHAIN_VARIANT not set, trying to locate Zephyr SDK
-- Found host-tools: zephyr 0.17.0 (/home/freddie/Workspaces/zephyr-sdk-0.17.0)
-- Found toolchain: zephyr 0.17.0 (/home/freddie/Workspaces/zephyr-sdk-0.17.0)
-- Found Dtc: /home/freddie/Workspaces/zephyr-sdk-0.17.0/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.6.0", minimum required is "1.4.6") 
-- Found BOARD.dts: /home/freddie/Workspaces/zephyrproject/zephyr/boards/raspberrypi/rpi_pico/rpi_pico.dts
-- Found devicetree overlay: /home/freddie/Workspaces/zephyrproject/zephyr/boards/shields/rpi_pico_uno_flexypin/rpi_pico_uno_flexypin.overlay
-- Found devicetree overlay: /home/freddie/Workspaces/zephyrproject/zephyr/boards/shields/nrf7002ek/nrf7002ek.overlay
-- Found devicetree overlay: /home/freddie/Workspaces/zephyrproject/examples/blinky/boards/rpi_pico.overlay
devicetree error: child specifier for <Node /connector in /home/freddie/Workspaces/zephyrproject/zephyr/boards/raspberrypi/rpi_pico/rpi_pico-common.dtsi:30> (b'\x00\x00\x00\x1b\x00\x00\x00 ') does not appear in <Property 'gpio-map' on /connector in /home/freddie/Workspaces/zephyrproject/zephyr/boards/shields/rpi_pico_uno_flexypin/rpi_pico_uno_flexypin.overlay:15>
CMake Error at /home/freddie/Workspaces/zephyrproject/zephyr/cmake/modules/dts.cmake:306 (execute_process):
  execute_process failed command indexes:

    1: "Child return code: 1"

Call Stack (most recent call first):
  /home/freddie/Workspaces/zephyrproject/zephyr/cmake/modules/zephyr_default.cmake:131 (include)
  /home/freddie/Workspaces/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:66 (include)
  /home/freddie/Workspaces/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:92 (include_boilerplate)
  CMakeLists.txt:4 (find_package)


-- Configuring incomplete, errors occurred!
FATAL ERROR: command exited with status 1: /usr/bin/cmake -DWEST_PYTHON=/home/freddie/Workspaces/zephyrproject/.venv/bin/python3 -B/home/freddie/Workspaces/zephyrproject/examples/blinky/build -GNinja -DBOARD=rpi_pico '-DSHIELD=rpi_pico_uno_flexypin;nrf7002ek' -S/home/freddie/Workspaces/zephyrproject/examples/blinky
```

### Impact

Not sure

### Environment

OS: Linux
Zephyr commit ID: 18d885d76cc20183b820d2aac8127464dabe9521
Toolchain: zephyr-sdk-0.17.0

### Additional Context

_No response_

--- a/boards/shields/rpi_pico_uno_flexypin/rpi_pico_uno_flexypin.overlay
+++ b/boards/shields/rpi_pico_uno_flexypin/rpi_pico_uno_flexypin.overlay
@@ -7,7 +7,7 @@
 #include <zephyr/dt-bindings/gpio/arduino-header-r3.h>
 
 / {
-	arduino_header: connector {
+	arduino_header: arduino_connector {
 		compatible = "arduino-header-r3";
 		#gpio-cells = <2>;
 		gpio-map-mask = <0xffffffff 0xffffffc0>;

